### ISSUE TITLE
False Positive: Application Error Disclosure (for JavaScript code files including error messages)

### ISSUE THREAD

**Issue by brittag** [zaproxy/zaproxy#4210]
My team uses a library that includes the file at https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.35/pdfmake.min.js (note: large file), and OWASP ZAP gives us the following Application Error Disclosure false positive:

```
URL    https://EXAMPLE.COM/pdfmake.min.js
Method    GET
Evidence    internal error
```

This is a false positive because this isn’t an application error page. Instead, it’s a large JavaScript file that includes the following code to generate error messages:

```
if(!(e<=-10))throw new Error("iconv-lite internal error: invalid decoding table value "+e+" at "+r+"/"+p);
```

To prevent this false positive, ZAP could look for evidence that a file is a code file instead of an error page. For example, `content-type: application/javascript` is likely to indicate a code file instead of an error page.

Thank you!

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#1230.

**Comment by brittag** [comment]
Nice, thanks!
